### PR TITLE
refactor lib, standardize koop

### PR DESCRIPTION
**Pull request by ngoldman**
Porting over changes from https://github.com/koopjs/koop-lib and doing a bit more (and a bit less). Once this is vetted and merged I'll delete the koop-lib repo. See https://github.com/koopjs/koop-lib/issues/1 for more context.

I originally went a bit farther than this but decided it's better to maintain the current lib API since it's exposed and used elsewhere. All tests should be passing and this pass shouldn't break anything. Mostly clean up, reformatting for standard, and trying to catch more errors.

@chelm @dmfenton @jgravois code review and feedback welcome.

### CODE REVIEW

File: lib/Exporter.js [hunk 1]
```diff
@@ -1,25 +1,22 @@
...
+var crypto = require('crypto')
+var projCodes = require('esri-proj-codes')
+var mv = require('mv')
+var spatialReference = require('./SpatialReference')
+var rm = require('rimraf')
+var exec = require('child_process').exec
 var ogrFormats = {
   kml: 'KML',
-  zip: '\"ESRI Shapefile\"',
+  zip: '"ESRI Shapefile"',
```
Comment (ngoldman) [inline lib/Exporter.js]: I think this got automatically stripped out by a linter -- are the backslashes needed?

Comment (dmfenton) [inline lib/Exporter.js]: @ngoldman I believe so as these formats end up as part of command line syntax

**Comment by dmfenton** [comment]
Just FYI as I read through this I'm marking a lot of things that should probably change. It's fine if they are out of scope for this PR

**Comment by dmfenton** [comment]
I don't see any showstoppers in here and I don't think we should overload this PR by refactoring every single thing. So I'm good with this. Thanks for doing the dirty work @ngoldman

File: lib/Exporter.js (hunk 1 above)
Comment (ngoldman) [inline lib/Exporter.js]: This looks like something https://github.com/maxogden/standard-format is doing via `standard -F`. Reported the issue here https://github.com/maxogden/standard-format/issues/84. I'll add them back in.

**Comment by ngoldman** [comment]
Thanks for the code review @dmfenton. We should make new issues to cover what you noticed.

**Comment by ngoldman** [comment]
Since tests are passing and @dmfenton gave it the OK I'm going to merge.

File: lib/Exporter.js (hunk 1 above)
Comment (feross) [inline lib/Exporter.js]: I think these backslashes are doing nothing. The backslash is the escape character, so if you want it to stay in the string, you need to write `\\`.

You can verify this in any Javascript console:

```
$ node

> '\"ESRI Shapefile\"'
'"ESRI Shapefile"'

> '"ESRI Shapefile"'
'"ESRI Shapefile"'
```

Comment (feross) [inline lib/Exporter.js]: If you wanted to use `"` instead of `'`, then you would need the backslashes.

```
> "\"ESRI Shapefile\""
'"ESRI Shapefile"'
```

Comment (chelm) [inline lib/Exporter.js]: Yup these are not needed. I'm making PR to remove them and I added a test to make all is good.

**Pull request merged**